### ISSUE TITLE
Meta: Use case documentation

### ISSUE THREAD

**Issue by pmorie** [kubernetes-sigs/service-catalog#20]
As discussed in the 20161024 SIG meeting, we should finalize the use cases for the v1 milestone of service catalog.

Use cases articulated during the meeting:
1.  Adding service brokers into catalog
2.  Catalog curation: which broker-provided services are globally visible?
3.  Catalog curation: which namespaces can see which catalog services?
4.  As a consumer of a service, how do I indicate that I want to consume a particular service?
5.  As a consumer of a service, how do I indicate that I want to bind to an instance of a service?
6.  What is the unit of consumption of a service?  Namespace?  Pod? Something else? (brian to comment)
7.  What is the versioning and update story for a service: example: what happens when a broker changes the services it provides?
8.  When a user binds to an instance of a service, what does that look like?  Which resources are created in their namespace?
9.  What is the update story for bindings to a service instance?
10. How does the catalog support multiple consumers in different Kubernetes namespaces of the same instance of a service? 
11. How do I control the number and identity of consumers allowed to bind to my instance?
12. Curation: how do I control who can see a service instance that I have provisioned?
13. Curation: how do I control the number of consumers of my service instance?

**Comment by westonh** [comment]
As an addendum to 1. Is there an auth story for adding service brokers? Presumably yes since we want to support private catalogs.

**Comment by pmorie** [comment]
Arguably most important, as @bgrant0607 pointed out in his post to the mailing list, what does consuming a service look like to a user?

**Comment by westonh** [comment]
A few others:

- Does my catalog list not only my service types but also my instances? E.g., I have a dev cluster running on my local machine and I want to connect my application with a database that exists in a test cluster. How do I find that database instance? 

- Importing a list of brokers: If I have a remote cluster with a service controller that has _n_ brokers registered, how can I import those into my local cluster? Related to the above, can I export a list of service brokers and types from my service controller?

**Comment by slack** [comment]
Would it be helpful to group these use cases? Something like:

## Catalog Publishing/Curation/Discovery

* (1) Adding service brokers into catalog
* (2) Catalog curation: which broker-provided services are globally visible?
* (3) Catalog curation: which namespaces can see which catalog services?
* How are services identified: name, service name/id, plan name/id?

## Requesting Services (Consumer)

* (4) As a consumer of a service, how do I indicate that I want to consume a particular service?
* (5) As a consumer of a service, how do I indicate that I want to bind to an instance of a service?
* (10) How does the catalog support multiple consumers in different Kubernetes namespaces of the same instance of a service?

## Provisioning/Deprovisioning

* (13) Curation: how do I control the number of consumers of my service instance?

## Binding/Unbinding

* (8) When a user binds to an instance of a service, what does that look like? Which resources are created in their namespace?
* (11) How do I control the number and identity of consumers allowed to bind to my instance?
* (12) Curation: how do I control who can see a service instance that I have provisioned?

## Using the Service (Consumption)

* (6) What is the unit of consumption of a service? Namespace? Pod? Something else? (brian to comment)

## Lifecycle of Service

* (9) What is the update story for bindings to a service instance?
* (7) What is the versioning and update story for a service: example: what happens when a broker changes the services it provides?

**Comment by westonh** [comment]
+1, this is great.

**Comment by slack** [comment]
Comments on this issue have been folded into #23.

**Comment by pmorie** [comment]
Reopening, because I think there is another set of use-cases we haven't talked through: use-cases for producing services (ie, how do I run my own service broker)

**Comment by bgrant0607** [comment]
@pmorie Aren't we just focusing on consumption of blackbox services at this point?

**Comment by pmorie** [comment]
Someone needs to take a pass back through this stuff and make sure it's done.  I believe it is, but I don't have time to check at the moment.  I'm going to move this to MVP 2 because it really needs to be done before Kubecon, not at MVP 1.

**Comment by arschles** [comment]
I have designated this a meta issue and removed it from a milestone

**Comment by mrbobbytables** [comment]
This project is being archived, closing open issues and PRs. 
Please see this PR for more information: https://github.com/kubernetes/community/pull/6632

/close
